mozilla-release/mobile/android/base/java/org/mozilla/gecko/home/BookmarksPanel.java

```diff
@@ -84,6 +84,7 @@
 
     @Override
     public void restoreData(@NonNull Bundle data) {
+        /* Cliqz Start o/
```

Comment:
I think we need the data to be restored when the mobile rotates as well.
Try to debug how they handle the back button in case you are inside a folder and follow it when you back from a page

Reply:
Hi Moaz! This method is not called on configuration change.  Restoring the bookmarks data on configuration change takes place in the `onConfigurationChanged`  of this fragment itself.